nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/scalar/comparison/ScalarGreaterThan.java

```diff
@@ -56,7 +58,7 @@ public ScalarGreaterThan(INDArray x, INDArray z, Number num) {
     }
 
     public ScalarGreaterThan(INDArray x, Number num) {
-        super(x, num);
+        this(x, Nd4j.createUninitialized(DataType.BOOL, x.shape(), x.ordering()), num);
```

Comment:
Let's just pass null for result array, rather than creating it here. If that doesn't work, that's something that should be flagged + fixed.
Same for the rest of the ops like this.